core/kazoo_documents/src/kzd_box_message.erl

```diff
@@ -74,21 +74,23 @@
 %%--------------------------------------------------------------------
 -spec new(ne_binary(), kz_proplist()) -> doc().
 new(AccountId, Props) ->
-    {Year, Month, _} = erlang:date(),
+    UtcSeconds = props:get_value(<<"Message-Timestamp">>, Props, kz_util:current_tstamp()),
+    Timestamp  = props:get_value(<<"Document-Timestamp">>, Props, UtcSeconds),
+    {Year, Month, _} = kz_util:to_date(Timestamp),
+    MediaId = props:get_value(<<"Media-ID">>, Props, kz_util:rand_hex_binary(16)),
     Db = kazoo_modb:get_modb(AccountId, Year, Month),
-    MediaId = <<(kz_util:to_binary(Year))/binary
-                ,(kz_util:pad_month(Month))/binary
-                ,"-"
-                ,(kz_util:rand_hex_binary(16))/binary
-              >>,
+    MsgId = <<(kz_util:to_binary(Year))/binary
+              ,(kz_util:pad_month(Month))/binary
+              ,"-"
+              ,MediaId/binary
+            >>,
```

Comment:
https://github.com/2600hz/kazoo/blob/master/core/kazoo/include/kz_types.hrl#L326 should be used to create modbs. BTW this one is PREFIX, are you instead looking for SUFFIX?